GitHubCore.ps1

```diff
@@ -989,3 +1070,98 @@ function Get-MediaAcceptHeader
 
     return $resultHeaders
 }
+
+@{
+    defaultJsonBodyContentType = 'application/json; charset=UTF-8'
+    defaultInFileContentType = 'text/plain'
+
+    # Compiled mostly from https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types
+    extensionToContentType = @{
```

Comment:
Very good questions:
  1. No idea.  Their documentation is unclear on this.  That bring up a good point though.  I have telemetry in place to [capture the extension if it was unknown](https://github.com/microsoft/PowerShellForGitHub/blob/0f9278bceab032324d530e0841a36f411f2fe59e/GitHubCore.ps1#L242) so that it could be added specifically, but it does look like I should add telemetry for the extension, regardless, so that we have it in the case of a failed upload.
  2. Another good question that I don't have an answer to.  I'm hoping that the added telemetry for all extensions as described in #1 will help with making that known when looking at failures for this method.

Reply:
>  have telemetry in place to capture the extension if it was unknown so that it could be added specifically

Yeah I saw that! Awesome idea!

I ask the questions just because they occurred to me, but I think your PR gets it right; try the accepted content-type, allow override, track if it doesn't work so it can be fixed. GitHub may be a moving target, so I can imagine trying to be too clever (like by rejecting ahead of time) might backfire

Reply:
Agreed.  I think for now, I'll keep the logic as-is (with the additional change of always capturing the file's extension).